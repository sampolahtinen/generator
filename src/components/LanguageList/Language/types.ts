export interface LanguageProps {
  title: string;
  percentage: number;
}
