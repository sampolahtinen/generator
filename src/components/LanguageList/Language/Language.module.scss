@import '../../../styles/colors.scss';

.language {
  margin-bottom: 2rem;
  white-space: nowrap;

  .titleWrapper {
    margin-bottom: 1.2rem;
  }

  .percentage {
    color: $grey_300;
    margin-left: 0.8rem;
  }
}
