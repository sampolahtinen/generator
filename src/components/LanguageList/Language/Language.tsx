import React from 'react';
import styles from './Language.module.scss';
import BarGauge from '../../BarGauge';
import { LanguageProps } from './types';

const Language = ({ title, percentage }: LanguageProps) => {
  return (
    <div className={styles.language}>
      <div className={styles.titleWrapper}>
        <span className="highlighted-text">{title}</span>
        <span className={styles.percentage}>{percentage}%</span>
      </div>
      <BarGauge value={percentage} />
    </div>
  );
};

export default Language;
