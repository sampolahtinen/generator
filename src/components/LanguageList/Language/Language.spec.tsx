import React from 'react';
import { shallow } from 'enzyme';
import Language from './Language';

describe('Language Component', () => {
  const wrapper = shallow(<Language title="JavaScript" percentage={50} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a title', () => {
    expect(wrapper.find('.highlighted-text').text()).toEqual('JavaScript');
  });

  it('displays a percentage', () => {
    expect(wrapper.find('.percentage').text()).toEqual('50%');
  });
});
