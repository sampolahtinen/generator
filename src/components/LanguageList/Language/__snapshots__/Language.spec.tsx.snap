// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Component renders 1`] = `
<div
  className="language"
>
  <div
    className="titleWrapper"
  >
    <span
      className="highlighted-text"
    >
      JavaScript
    </span>
    <span
      className="percentage"
    >
      50
      %
    </span>
  </div>
  <BarGauge
    value={50}
  />
</div>
`;
